tests/test_glean.py

```diff
@@ -215,6 +215,93 @@ def test_pings(self, glean):
             }
             assert glean.get_ping_descriptions() == {"foo": "baz"}
 
+    def test_dependencies(self, glean):
+        RETURN_VALUES = {"glean-core": {"name": "glean-core", "type": "dependency"}}
+        with patch.object(
+            glean,
+            "_get_json",
+            return_value=RETURN_VALUES,
+        ):
+            assert set(glean.get_dependencies()) == {
+                "glean-core",
+            }
+
+    def test_probes(self, glean):
+        RETURN_VALUES = {
+            "test.probe": {
+                "history": [
+                    {
+                        "dates": {
+                            "first": "2023-11-27 21:59:53",
+                            "last": "2024-01-05 17:36:15",
+                        },
+                        "description": "description",
+                        "expires": "never",
+                        "type": "string",
+                        "version": 0,
+                    }
+                ],
+                "in-source": True,
+                "name": "test.probe",
+                "type": "string",
+            }
+        }
+
+        with patch.object(glean, "get_dependencies", return_value=[]):
+            with patch.object(
+                glean,
+                "_get_json",
+                return_value=RETURN_VALUES,
+            ):
+                probes = glean.get_probes()
+                assert len(probes) == 1
+                assert probes[0].id == "test.probe"
+                assert probes[0].type == "string"
+
+    def test_probes_multiple_types(self, glean):
+        RETURN_VALUES = {
+            "test.probe": {
+                "history": [
+                    {
+                        "dates": {
+                            "first": "2023-11-27 21:59:53",
+                            "last": "2024-01-05 17:36:15",
+                        },
+                        "description": "description",
+                        "expires": "never",
+                        "type": "string",
+                        "version": 0,
+                    },
+                    {
+                        "dates": {
+                            "first": "2024-01-05 21:59:53",
+                            "last": "2024-01-06 17:36:15",
+                        },
+                        "description": "description",
+                        "expires": "never",
+                        "type": "url",
+                        "version": 0,
+                    },
+                ],
+                "in-source": True,
+                "name": "test.probe",
+                "type": "string",
```

Comment:
Ok, great. I typed up a whole response here and then GitHub just didn't save that or something.

So here it is again:
Initially I thought this should be `"type": "url"`, assuming that probeinfo service would do the right thing already.
But it does not: https://probeinfo.telemetry.mozilla.org/glean/moso-mastodon-web/metrics
Look for `page_url`.

The dictionary does list that metrics as URL though, so I guess it picks the last seen value:
https://dictionary.telemetry.mozilla.org/apps/moso_mastodon_web/metrics/web_page_url

Changing probeinfo then is followup work I guess

Reply:
I created an issue for this here: https://github.com/mozilla/probe-scraper/issues/677